src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -266,6 +319,9 @@ protected override ModuleFieldLayout CreateValueFromKey(EcmaModule module)
                         break;
 
                     case CorElementType.ELEMENT_TYPE_VAR:
+                        GetElementTypeInfoGeneric(module, fieldDesc, valueTypeHandle, moduleLayout, ref alignment, ref size, ref isGcPointerField, ref isGcBoxedField);
```

Comment:
Would it make sense to have this called from line 435 if the owning type is generic (i.e. use the type system instead of metadata for all fields on generic types, not just those of type `T`)?

This is just to make sure the code gets regularly exercised. The way the new logic gets called right now makes pieces of it run only very rarely and we might not get much coverage for it (e.g. there's pieces of logic that would only run if a generic type is instantiated over an enum and it has a static field of type T).